### ISSUE TITLE
[suggestion] asyncify your callbackicity :D

### ISSUE THREAD

**Issue by weisk** [nikensss/nodejs_design_patterns#1]
the following signature is the conventional 'callback pattern'

```js
readFile(file, 'utf8', (err, content) => {
```

which we can 'coolify' with util.promisify (node builtin library)

```js
import { promisify } from 'util';
import { readFile } from 'fs';

const read = promisify(readFile);
```

now `read` returns a promise, so we can do this now:

```js
async function find(file) {
  try {
    const content = await read(file);
    console.log(content);
  } catch (err) {
    throw err;
  }
}
```

**Comment by nikensss** [comment]
Promises come later in the book. It first covers callbacks and EventEmmiter to later on move to promises.

Also, wouldn't it be better to use fs.promises for that rather than promisifying?

**Comment by weisk** [comment]
yeah well this is just something that im explaining to you, given that you
implemented it in a callback style manner.

Any library function working with that style is really easy to 'tweak' to
async style.

As per this particular case, I think I would just use a file stream.

**Comment by nikensss** [comment]
Ah, I see. The book also covers streams later on. So I guess this book does have a nice structure! :D
